src/main/scala/controllers/ApplicationController.scala

```diff
@@ -38,44 +38,32 @@ class ApplicationController @Inject()(actionHandler: ActionHandler, applications
   def editSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
     actionHandler.gatherSectionDetails(id, sectionNumber).flatMap {
       case Some((app, af, afs, opp)) =>
-        fieldsFor(sectionNumber) match {
-          case Some(fields) =>
-            applications.getSection(id, sectionNumber).flatMap { section =>
-              val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
-              actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, fields, noErrors, hints)
-            }
-          case None => Future(NotFound)
+        applications.getSection(id, sectionNumber).flatMap { section =>
+          val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
+          actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, noErrors, hints)
         }
       case None => Future(NotFound)
     }
   }
 
   def resetAndEditSection(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
-    fieldsFor(sectionNumber) match {
-      case Some(fields) =>
-        applications.clearSectionCompletedDate(id, sectionNumber).map { _ =>
-          Redirect(controllers.routes.ApplicationController.editSectionForm(id, sectionNumber))
-        }
-      case None => Future(NotFound)
+    applications.clearSectionCompletedDate(id, sectionNumber).map { _ =>
+      Redirect(controllers.routes.ApplicationController.editSectionForm(id, sectionNumber))
     }
   }
 
   def showSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
     actionHandler.gatherSectionDetails(id, sectionNumber).flatMap {
       case Some((overview, form, afs, opp)) =>
-        fieldsFor(sectionNumber) match {
-          case Some(fields) =>
-            applications.getSection(id, sectionNumber).flatMap { section =>
-              section.flatMap(_.completedAtText) match {
-                case None =>
-                  val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
-                  actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, fields, noErrors, hints)
-                case _ =>
-                  Future.successful(actionHandler.redirectToPreview(id, sectionNumber))
-
-              }
-            }
-          case None => Future(NotFound)
+        applications.getSection(id, sectionNumber).flatMap { section =>
+          section.flatMap(_.completedAtText) match {
+            case None =>
+              val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
```

Comment:
I've replaced a lot of `List()` with `List.empty`. Also `Seq()` with `List.empty` which, again, should be a little bit faster.